### ISSUE TITLE
[HOLD #27977][$500] Chat - Cursor is not displayed after 6 lines have been entered in compose box

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#35369]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**  v1.4.33-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
Pre-requisite: user must be logged in.

1. Go to any chat.
2. Tap on the compose box.
3. Tap on Enter until you have entered more than 6 lines.
## Expected Result:
The cursor should be displayed no matter how many lines are being entered.

## Actual Result:
The cursor disappears after more than 6 lines have been entered.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/d27d0449-d8f6-4ee8-ad6d-6369ad2663cb

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dd5527568c5f7f46</li>
        <li>Upwork Job ID: 1752276854903971840</li>
        <li>Last Price Increase: 2024-01-30</li>
    </ul>
</details>

**Comment by rayanetech** [comment]
I would like to work on this

**Comment by aswin-s** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Cursor goes out of view after adding more than 6 lines

### What is the root cause of that problem?

For small screen devices, the max number of lines for Composer input is set to 6 lines. On mobile safari beyond this height overflow scroll behaviour sets in. So even though the scroll height of TextArea increases the visible area of the TextArea is limited to 6 lines. So cursor moves beyond the visible area when new line is entered beyond 6 lines. That's the reason cursor gets hidden.

However when user enters any other character the scroll position is reset to the cursor position making the cursor visible again.



### What changes do you think we should make in order to solve the problem?

The scroll position of the Composer TextArea should be set to the cursor position at all times, even when the character entered is a new line. To acheive this we already have a hook `updateMultilineInputRange` that sets scrollTop to current scroll height of input. But for ComposerWithSuggestion it gets invoked only on mount as the dependency array of useEffect hook is set to `[]`.

https://github.com/Expensify/App/blob/4e88eea26a79c7d69e3bf9fac28a515586c3855b/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js#L544-L555

Instead add `value` to dependency array so that the scroll position is set whenever value changes. This ensures that scroll position is maintained even when new line is entered.


```tsx
useEffect(() => {
  // Scrolls the composer to the bottom and sets the selection to the end, so that longer drafts are easier to edit
  updateMultilineInputRange(textInputRef.current, shouldAutoFocus);

  if (value.length === 0) {
    return;
  }

  Report.setReportWithDraft(reportID, true);
  
  // Add value as a dependency so that the hook runs on value change
  // eslint-disable-next-line react-hooks/exhaustive-deps
}, [value]); 
```

Result

| iOS | Web |
|--|--|
| <video src="https://github.com/Expensify/App/assets/6880914/b4c468ec-025b-465c-bfa8-b684d2f9cb46" /> | <video src="https://github.com/Expensify/App/assets/6880914/84efe3ae-980d-483a-84c1-ff99911665ae" /> |

### What alternative solutions did you explore? (Optional)

None

**Comment by mananjadhav** [comment]
@aswin-s's proposal seems to be working.  I can see both the videos are iOS native, @aswin-s but the header says `Web` on the right.

🎀 👀 🎀 C+ Reviewed.

**Comment by aswin-s** [comment]
> @aswin-s's proposal seems to be working.  I can see both the videos are iOS native, @aswin-s but the header says `Web` on the right.
> 
> 
> 
> 🎀 👀 🎀 C+ Reviewed.

@mananjadhav Ah my bad! It was supposed to be before and after videos. Both videos are for web. I took it from Chrome device simulator. Will fix the titles shortly.

**Comment by thienlnam** [comment]
Looks like this was created before live markdown was added - let's see if this is still reproducible first and then if it is, it would likely be independent of the live markdown changes and can be addressed seperately

**Comment by mananjadhav** [comment]
@kavimuru Based on the previous comment, can you please retest this?

**Comment by aswin-s** [comment]
MarkdownTextInput was added only for native platforms. This is still reproducible on main.

**Comment by danieldoglas** [comment]
Yes, but we're also merging the new markdown on web soon. Let's wait on that before trying to solve this issue. moving this to weekly until then.

**Comment by mananjadhav** [comment]
@danieldoglas @thienlnam Do we have any PR or issue that we can link against?

**Comment by danieldoglas** [comment]
The web implementation was merged yesterday in the react-live-markdown repo https://github.com/Expensify/react-native-live-markdown/pull/32, and it looks like it's making it's way to web here https://github.com/Expensify/App/pull/35837

**Comment by thienlnam** [comment]
Live markdown web has been merged!

**Comment by mananjadhav** [comment]
@kavimuru Is this still reproducible?

**Comment by thienlnam** [comment]
We actually ended up reverted the PR, but I'm going to update the issue it's held on so we can wait until then

**Comment by danieldoglas** [comment]
still on hold

**Comment by garrettmknight** [comment]
Looks like it's fixed, closing!
